packages/migration/src/migrations/user-mgnt/20231106130956-email-mobile-unique-index.ts

```diff
@@ -15,6 +15,40 @@ export const up = async (db: Db, client: MongoClient) => {
   const session = client.startSession()
   try {
     await session.withTransaction(async () => {
+      console.log('Checking if mobile or email field already have indices')
+      const indexes = Object.keys(
+        (await db.collection('users').stats()).indexSizes
+      )
+      const mobileHasIndex = indexes.some(
+        (index) => index.split('_')[0] === ('mobile' as const)
+      )
+      const emailForNotificationHasIndex = indexes.some(
+        (index) => index.split('_')[0] === ('emailForNotification' as const)
+      )
+
+      if (mobileHasIndex) {
+        console.log('Removing index of mobile')
+        await db.collection('users').dropIndex({ mobile: 1 } as any)
+      }
+
+      if (emailForNotificationHasIndex) {
+        console.log('Removing index of email')
+        await db
+          .collection('users')
+          .dropIndex({ emailForNotification: 1 } as any)
+      }
+      // Ensure that the "mobile" field in the "users" collection does not contain null or duplicate values
+      // Before making the field sparse, which allows it to have null values but ensures uniqueness
+      const updateResult = await db.collection('users').updateMany(
+        { mobile: { $in: [null, ''] } },
```

Comment:
Should empty string actually be a valid value here? I'm thinking if you create a idea and leave the field empty, then it would represent that it was explicitly left empty. Null or the field not existing would represent the value for the field was never supplied. Null is probably a good default for this so all the user objects in the database would be structurally similar.

Reply:
Thanks for the explanation. Yea, I think it's smart to consolidate the possible values to there being either a non-empty string or no field at all.  One question comes to mind – if we assume (or know) there have been nulls and empty strings in the data before, should we investigate where those came from? If it's still possible to submit null or an empty string, then this issue would start happening again, right?